dts/bindings/sensor/st,stm32-qdec.yaml

```diff
@@ -70,3 +70,18 @@ properties:
     description: |
       This is a number >= 1 that is used to determine how many revolutions
       were done based on the current counter's value.
+
+  st,encoder-mode:
+    type: int
+    description: |
+      Encoder counting mode
+      MODE_X2_TI1 Counter counts up/down on TI2FP2 edge depending on TI1FP1 level
+      MODE_X2_TI2  Counter counts up/down on TI1FP1 edge depending on TI2FP2 level
+      MODE_X4_TI12 Counter counts up/down on both TI1FP1 and TI2FP2 edges
+      depending on the level of the other input
+      Defaults on MODE_X2_TI1 since it is the low level driver default
```

Comment:
Doesn't look like a right justification (and for sure not to the reader).
Isn't it the HW default reset value ?

Reply:
No after reset encoder mode is not enabled. You have to program if you want to use encoders.
The default I chose is 
1- The actual behavior of the driver already in mainline
2- The default of the Low Level Driver.

Reply:
Btw, here is the reason why we're bothering you with this: https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#rules-for-default-values

In that case, setting a default value wouldn't break anything, but I'd say that this is now an application choice, with no reason to default to one or the other (except the legacy behavior).

The only good reason would be that with this change, not providing default would break existing users. But I think we should rather try to detect if missing and raise an error. Setting it as `required: true` would allow to do that, and I think that should be sufficient. Since it would be a breaking change, this would require to be documented in release notes.

@MaureenHelm What would be you opinion?

Reply:
I see. This is my first attempt at a contribution and I am trying to do it because I thought it was a good idea to give back this super tiny bit; that said I tried to follow the guidelines but since I am doing it in my spare time I haven´t studied all the possible documents.

Now I am leaving for a couple of weeks, so I won't be able to follow up, so if you want to shelve this no problem, maybe I will try again in the future.

Reply:
The default value in the binding should correspond to the hardware default. It's unfortunate that the LL driver default doesn't match, so in this case I think it makes sense to make the property required to avoid quietly changing driver behavior.

Reply:
Hi @erwango @MaureenHelm I pushed the modification to change the field from defaulted to required.